xtable-core/src/main/java/org/apache/xtable/delta/DeltaValueConverter.java

```diff
@@ -194,9 +194,35 @@ public static Object convertFromDeltaPartitionValue(
 
   private static Object castObjectToInternalType(Object value, InternalType valueType) {
     switch (valueType) {
+      case DOUBLE:
+        if (value instanceof String)
+          switch (value.toString()) {
+            case "NaN":
```

Comment:
Is there anywhere we can import these strings as constants from Delta? if not, can you define some internally for us?

Reply:
After searching there was no constants defined in Delta and will define them internally. 
How about defining them inside DeltaValueConverter class like
```
public class DeltaValueConverter {
  private static final String DATE_FORMAT_STR = "yyyy-MM-dd HH:mm:ss";
  private static final TimeZone TIME_ZONE = TimeZone.getTimeZone("UTC");
  protected static final String NAN_VALUE = "NaN";
  protected static final String INFINITY_VALUE = "Infinity";
...
```

Reply:
Thanks, looks good to me!